staging/src/k8s.io/apiserver/pkg/audit/context.go

```diff
@@ -90,7 +91,15 @@ func AddAuditAnnotations(ctx context.Context, keysAndValues ...string) {
 	defer ac.annotationMutex.Unlock()
 
 	if len(keysAndValues)%2 != 0 {
+		// Option 1: (original) Don't touch unstructured log lines.
 		klog.Errorf("Dropping mismatched audit annotation %q", keysAndValues[len(keysAndValues)-1])
+
+		// Option 2: Munge with format string
+		logger := klog.FromContext(ctx)
+		logger.Error(nil, fmt.Sprintf("Dropping mismatched audit annotation %q", keysAndValues[len(keysAndValues)-1]))
+
+		// Option 3: Rewrite as structured logs
+		logger.Error(nil, "Dropping mismatched audit annotation", "annotation", keysAndValues[len(keysAndValues)-1])
```

Comment:
With my structured logging WG lead hat on I have to vote for option 3 :grin: 

The resulting messages will be different, but still have the same information. If you stay consistent, then searching for key/value pairs becomes easier, even when using the text output format.

When using JSON, the overhead for writing log entries is lower because zap is more optimized than klog.